org.eclipse.ice.dev/org.eclipse.ice.dev.jsonschemaconverter/src/main/java/org/eclipse/ice/dev/jsonschemaconverter/JsonSchemaConverter.java

```diff
@@ -0,0 +1,427 @@
+    * @param input string to check
+    * @return boolean representing whether or not input is a float
+    */
+   public static boolean isFloat(String input) {
+	   final String Digits     = "(\\p{Digit}+)";
+	   final String HexDigits  = "(\\p{XDigit}+)";
+	   // an exponent is 'e' or 'E' followed by an optionally
+	   // signed decimal integer.
+	   final String Exp        = "[eE][+-]?"+Digits;
+	   final String fpRegex    =
+	       ("[\\x00-\\x20]*"+  // Optional leading "whitespace"
+	        "[+-]?(" + // Optional sign character
+	        "NaN|" +           // "NaN" string
+	        "Infinity|" +      // "Infinity" string
+
+	        // A decimal floating-point string representing a finite positive
+	        // number without a leading sign has at most five basic pieces:
+	        // Digits . Digits ExponentPart FloatTypeSuffix
+	        //
+	        // Since this method allows integer-only strings as input
+	        // in addition to strings of floating-point literals, the
+	        // two sub-patterns below are simplifications of the grammar
+	        // productions from section 3.10.2 of
+	        // The Java™ Language Specification.
+
+	        // Digits ._opt Digits_opt ExponentPart_opt FloatTypeSuffix_opt
+	        "((("+Digits+"(\\.)?("+Digits+"?)("+Exp+")?)|"+
+
+	        // . Digits ExponentPart_opt FloatTypeSuffix_opt
+	        "(\\.("+Digits+")("+Exp+")?)|"+
+
+	        // Hexadecimal strings
+	        "((" +
+	         // 0[xX] HexDigits ._opt BinaryExponent FloatTypeSuffix_opt
+	         "(0[xX]" + HexDigits + "(\\.)?)|" +
+
+	         // 0[xX] HexDigits_opt . HexDigits BinaryExponent FloatTypeSuffix_opt
+	         "(0[xX]" + HexDigits + "?(\\.)" + HexDigits + ")" +
+
+	         ")[pP][+-]?" + Digits + "))" +
+	        "[fFdD]?))" +
+	        "[\\x00-\\x20]*");// Optional trailing "whitespace"
+
+	   return Pattern.matches(fpRegex, input);
+   }
+
+   /**
+    * Checks if input string represents a boolean.
+    * @param input string to check
+    * @return boolean representing whether or not input is a boolean
+    */
+   public static boolean isBoolean(String input) {
+	   if (input == null) {
+		   return false;
+	   }
+	   return input.trim().equals("false") || input.trim().equals("true");
+   }
+
+   /**
+    * Checks if input string represents an int.
+    * @param input string to check
+    * @return boolean representing whether or not input is an int
+    */
+   public static boolean isInt(String input) {
+	   try {
+		   Integer.parseInt(input);
+		   return true;
+	   } catch (Exception e) {
+		   return false;
+	   }
+   }
+
+   /**
+    * Gives the string representation of the type of input string.
+    * Used when input string may represent an int, float, boolean, etc.
+    * @param input string that may represent a type other than string
+    * @return the type of the input as a string
+    */
+   public static String getTypeAsString(String input) {
```

Comment:
We might want to move these types of functions to some sort of Java Source formatter at some point. I had to do similar things with the `DefaultFields` class and having some of these available to templates or annotation processors may come in handy. I think here is good for now -- just taking note of it.

Reply:
Agreed.